app/services/targets/create_service.rb

```diff
@@ -11,13 +11,30 @@ def initialize(current_user, params)
     def call
       raise MaxAllowTargetsError unless targets.count < MAX_ALLOWED_TARGETS
 
-      targets.create!(params)
+      Target.transaction do
+        target = targets.create!(params)
+        find_match(target)
+        target
+      end
     end
 
     private
 
     def targets
       @targets ||= current_user.targets
```

Comment:
I don't think we need to use memoization here.

Reply:
This is used above, when the user can only create maximum targets